app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/add_tools/+page.svelte

```diff
@@ -1,183 +1,145 @@
-      }
-      if (!server_url.trim()) {
-        throw new Error("Server URL is required")
-      }
-      if (headers.length > 0) {
-        for (const header of headers) {
-          if (!header.key.trim()) {
-            throw new Error("Header name is required")
-          }
-          if (!header.value.trim()) {
-            throw new Error("Header value is required")
-          }
-        }
-      }
-
-      const headersObj = buildHeadersObject()
-
-      const { data, error: api_error } = await client.POST(
-        "/api/projects/{project_id}/connect_remote_mcp",
+  const sampleMcpServers: McpServer[] = [
+    {
+      name: "GitHub",
+      description:
+        "Adds ability to read repositories and code files, manage issues and PRs, analyze code, and automate workflow.",
+      server_url: "https://api.githubcopilot.com/mcp/",
+      headers: [
         {
-          params: {
-            path: {
-              project_id: $page.params.project_id,
-            },
-          },
-          body: {
-            name: name.trim(),
-            server_url: server_url.trim(),
-            headers: headersObj,
-            description: description.trim() || null,
+          key: "Authorization",
+          value: "",
+          placeholder: "Bearer {GITHUB_PERSONAL_ACCESS_TOKEN}",
+        },
+      ],
+      button_text: "Connect",
+    },
+    {
+      name: "Twelve Data",
+      description:
+        "Integrates with Twelve Data API to provide real-time quotes, historical OHLCV price data, and instrument metadata for stocks, forex pairs, and cryptocurrencies across global markets.",
+      server_url: "https://mcp.twelvedata.com/mcp/",
+      headers: [
+        {
+          key: "Authorization",
+          value: "",
+          placeholder: "apikey {TWELVE_DATA_API_KEY}",
+        },
+        {
+          key: "X-OpenAPI-Key",
+          value: "",
+          placeholder: "YOUR_OPENAI_API_KEY",
+        },
+      ],
+      button_text: "Connect",
+    },
+    {
+      name: "Postman Echo",
+      description: "Simple MCP Server to test MCP tool connections.",
+      server_url: "https://postman-echo-mcp.fly.dev/",
+      headers: [],
+      button_text: "Connect",
+    },
+  ]
+
+  let sections = [
+    {
+      category: "Sample Tools",
+      items: [
+        // TODO: Add more custom tool servers, pre-filled in.
+        ...sampleMcpServers.map((tool) => ({
+          ...tool,
+          on_click: () => connectRemoteMcp(tool),
+        })),
+      ],
+    },
+    {
+      category: "Custom Tools",
+      items: [
+        {
+          name: "Remote MCP Servers",
+          description:
+            "Connect to remote MCP servers to add tools to your project.",
+          href: `/settings/manage_tools/${$ui_state?.current_project_id}/add_tools/remote_mcp`,
+          button_text: "Connect",
+          on_click: () => {
+            goto(
+              `/settings/manage_tools/${$ui_state?.current_project_id}/add_tools/remote_mcp`,
```

Comment:
Try to never use current_project_id, we should kill it eventually. It might not be the project_id in the URL, in which case you have a bug. It's mostly for loading the /run page or fresh page loads of root url ("/").

You should use $page.params.project_id, which is the actual project ID from the URL. setting this at the top is a common pattern: `$: project_id = $page.params.project_id`

There's a few other places of this too, grab em all.